CMakeLists.txt

```diff
@@ -21,19 +21,56 @@ if (NOT CMAKE_BUILD_TYPE)
             "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif ()
 
+# Set build options
+option (BUILD_UFS "Build GOCART component for UFS" OFF)
+
 set (DOING_GEOS5 YES)
 
 # Should find a better place for this - used in Chem component
 set (ACG_FLAGS -v)
 set (FV_PRECISION R4R8)
 
+if (BUILD_UFS)
+# Ensure we build as 32-bit
+  message ("Force 32-bit build for GOCART")
+  if (CMAKE_Fortran_COMPILER_ID MATCHES "Intel")
+    string (REPLACE "-real-size 64" "" CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}")
+  elseif (CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
+    string (REPLACE "-fdefault-real-8" "" CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}")
+  endif()
+endif()
+
 if (NOT COMMAND esma)
     list (APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake@")
     include (esma)
 endif()
 
     ecbuild_declare_project()
 
+if (NOT Baselibs_FOUND)
+  # Find dependencies
+  if (NOT NetCDF_Fortran_FOUND)
+    find_package (NetCDF REQUIRED Fortran)
+  endif ()
+  add_definitions(-DHAS_NETCDF3)
+
+  if (NOT ESMF_FOUND)
+    find_package (ESMF REQUIRED)
+  endif ()
+
+  find_package (GFTL REQUIRED)
+  find_package (GFTL_SHARED REQUIRED)
+  find_package (PFLOGGER REQUIRED)
+  find_package (YAFYAML REQUIRED)
+endif ()
+
+if (IS_DIRECTORY "${PROJECT_SOURCE_DIR}/ESMF/Shared/MAPL@")
+  message (STATUS "Using local MAPL source tree: ${PROJECT_SOURCE_DIR}/ESMF/Shared/MAPL@")
+else ()
```

Comment:
Another build error, this time for building with the full `GEOSgcm`. Could you change this line to:
```
elseif (NOT Baselibs_Found)
```
This will prevent the find package from interfering with the `GEOSgcm` build.

Reply:
Note that with this change, I can confirm that this branch will build and zero-diff run GOCART legacy with the `GEOSgcm`.

Reply:
@WilliamJamieson - I'm a bit confused. MAPL is not bundled with Baselibs. Does `GEOSgcm` build places MAPL in a different directory? If the directory is the same (`GOCART/ESMF/Shared/MAPL@`), the current `CMakeLists.txt` should work (it worked on Orion).

Reply:
MAPL goes in a completely different directory when in the `GEOSgcm` thats the issue.

Reply:
@rmontuoro - This is fine with me.

Reply:
This appears to have been addressed by commit 5f69e7c.